tritium-metrics-jvm/src/main/java/com/palantir/tritium/metrics/jvm/JvmMetrics.java

```diff
@@ -190,6 +183,16 @@ protected Ratio getRatio() {
         });
     }
 
+    private static long totalHeapPlusNonHeap(MemoryMXBean memoryBean, ToLongFunction<MemoryUsage> longFunction) {
+        MemoryUsage heap = memoryBean.getHeapMemoryUsage();
+        MemoryUsage nonHeap = memoryBean.getNonHeapMemoryUsage();
+        if (heap == null && nonHeap == null) {
+            return -1;
+        }
+        return (heap == null ? 0 : longFunction.applyAsLong(heap))
+                + (nonHeap == null ? 0 : longFunction.applyAsLong(nonHeap));
+    }
```

Comment:
I updated the target branch earlier this morning, no strong preference on which approach we merge though.
new commits:
https://github.com/palantir/tritium/pull/1392/commits/ad9ab1a684dd9d01e0535c4d0c9989a9738ae88d
https://github.com/palantir/tritium/pull/1392/commits/613e4e61a3aa888e2e94799b45ebbf8847e49dc9